diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x configure itcl3.2.1-orig/itcl/configure.in itcl3.2.1/itcl/configure.in
--- itcl3.2.1-orig/itcl/configure.in	2001-08-13 11:12:08.000000000 -0500
+++ itcl3.2.1/itcl/configure.in	2006-05-09 17:03:02.000000000 -0500
@@ -108,7 +108,11 @@
 #--------------------------------------------------------------------
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    CYGWIN*)
+	CYGPATH=echo
+	RELPATH=".. .. bin"
+    ;;
+    *win32* | *WIN32*)
 	CYGPATH="cygpath -w"
 	AC_CHECK_PROG(CYGPATH, cygpath, cygpath -w, echo)
 	RELPATH=".. .. bin"
@@ -131,7 +135,7 @@
 ITCL_UNIX_DIR_NATIVE=`${CYGPATH} ${srcdir}/unix`
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    *win32* | *WIN32*)
 	ITCL_PLATFORM_DIR_NATIVE=${ITCL_WIN_DIR_NATIVE}
     ;;
     *)
@@ -185,7 +189,14 @@
 #--------------------------------------------------------------------
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    CYGWIN*)
+	AC_DEFINE(BUILD_itcl)
+	CLEANFILES="*.dll.a *.dll *.exp"
+	PLATFORM_SOURCES='$(UNIX_SOURCES)'
+	PLATFORM_OBJECTS='$(UNIX_OBJECTS)'
+	PLATFORM_DIR='$(UNIX_DIR)'
+    ;;
+    *win32* | *WIN32*)
 	AC_DEFINE(BUILD_itcl)
 	CLEANFILES="*.lib *.dll *.exp *.ilk *.pdb vc50.pch"
 	PLATFORM_SOURCES='$(WIN_SOURCES)'
@@ -284,7 +295,17 @@
 #--------------------------------------------------------------------
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    CYGWIN*)
+	if test "${SHARED_BUILD}" = "1" ; then
+	    SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC} -Wl,--out-implib,lib${PACKAGE}${SHARED_LIB_SUFFIX}.a"
+	    eval "${PACKAGE}_LIB_FILE=cyg${PACKAGE}${SHARED_LIB_SUFFIX}"
+	    RANLIB=:
+	else
+	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${UNSHARED_LIB_SUFFIX}"
+	fi
+	eval "${PACKAGE}stub_LIB_FILE=lib${PACKAGE}stub${UNSHARED_LIB_SUFFIX}"
+	;;
+    *win32* | *WIN32*)
 	if test "${SHARED_BUILD}" = "1" ; then
 	    SHLIB_LD_LIBS="\"`cygpath -w ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\" ${TCL_SHLIB_LD_LIBS}"
 	    eval "${PACKAGE}_LIB_FILE=${PACKAGE}${SHARED_LIB_SUFFIX}"
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x configure itcl3.2.1-orig/itk/configure.in itcl3.2.1/itk/configure.in
--- itcl3.2.1-orig/itk/configure.in	2001-08-13 11:14:29.000000000 -0500
+++ itcl3.2.1/itk/configure.in	2006-05-09 17:03:55.000000000 -0500
@@ -104,7 +104,11 @@
 #--------------------------------------------------------------------
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    CYGWIN*)
+	CYGPATH=echo
+	RELPATH=".. .. bin"
+    ;;
+    *win32* | *WIN32*)
 	CYGPATH="cygpath -w"
 	AC_CHECK_PROG(CYGPATH, cygpath, cygpath -w, echo)
 	RELPATH=".. .. bin"
@@ -127,7 +131,7 @@
 ITK_UNIX_DIR_NATIVE=`${CYGPATH} ${srcdir}/unix`
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    *win32* | *WIN32*)
 	ITK_PLATFORM_DIR_NATIVE=${ITK_WIN_DIR_NATIVE}
     ;;
     *)
@@ -222,7 +226,14 @@
 #--------------------------------------------------------------------
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    CYGWIN*)
+	AC_DEFINE_UNQUOTED(BUILD_${PACKAGE})
+	CLEANFILES="*.dll.a *.dll *.exp"
+	PLATFORM_SOURCES='$(UNIX_SOURCES)'
+	PLATFORM_OBJECTS='$(UNIX_OBJECTS)'
+	PLATFORM_DIR='$(UNIX_DIR)'
+	;;
+    *win32* | *WIN32*)
 	AC_DEFINE_UNQUOTED(BUILD_${PACKAGE})
 	CLEANFILES="*.lib *.dll *.exp *.ilk *.pdb vc50.pch"
 	PLATFORM_SOURCES='$(WIN_SOURCES)'
@@ -323,7 +334,17 @@
 #--------------------------------------------------------------------
 
 case "`uname -s`" in
-    *win32* | *WIN32* | *CYGWIN_NT*|*CYGWIN_98*|*CYGWIN_95*)
+    CYGWIN*)
+	if test "${SHARED_BUILD}" = "1" ; then
+	    SHLIB_LD_LIBS="${tkstub_LIB_SPEC} ${TCL_BUILD_STUB_LIB_SPEC} ../itcl/${itclstub_LIB_FILE} -Wl,--out-implib,lib${PACKAGE}${SHARED_LIB_SUFFIX}.a"
+	    eval "${PACKAGE}_LIB_FILE=cyg${PACKAGE}${SHARED_LIB_SUFFIX}"
+	    RANLIB=:
+	else
+	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${UNSHARED_LIB_SUFFIX}"
+	fi
+	eval "${PACKAGE}stub_LIB_FILE=lib${PACKAGE}stub${UNSHARED_LIB_SUFFIX}"
+	;;
+    *win32* | *WIN32*)
 	if test "${SHARED_BUILD}" = "1" ; then
 	    SHLIB_LD_LIBS="\"`cygpath -w ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\" ${TCL_SHLIB_LD_LIBS} \"`${CYGPATH} ../itcl/${itclstub_LIB_FILE}`\" ${tkstub_LIB_SPEC} "
 	    eval "${PACKAGE}_LIB_FILE=${PACKAGE}${SHARED_LIB_SUFFIX}"
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x ltmain.sh -x 'config.*' -x depcomp -x install-sh -x missing -x mkinstalldirs -x autom4te.cache -x '*compile' -x 'intltool*.in' -x 'xml-i18n-*.in' -x '*.pyc' -x '*.mo' -x '*.gmo' -x ABOUT-NLS -x Makevars.template -x COPYING -x INSTALL -x '*.orig' -x '*.rej' -x '*~' -x '*.temp' -x configure itcl3.2.1-orig/tcl.m4 itcl3.2.1/tcl.m4
--- itcl3.2.1-orig/tcl.m4	2001-06-21 23:41:05.000000000 -0500
+++ itcl3.2.1/tcl.m4	2006-05-08 20:40:30.000000000 -0500
@@ -303,7 +303,7 @@
 	CC=gcc
     else
 	case "`uname -s`" in
-	    *win32* | *WIN32* | *CYGWIN_NT* | *CYGWIN_98* | *CYGWIN_95*)
+	    *win32* | *WIN32*)
 		CC=cl
 	    ;;
 	    *)
@@ -391,7 +391,7 @@
 	AC_DEFINE(_REENTRANT)
 
 	case "`uname -s`" in
-	    *win32* | *WIN32* | *CYGWIN_NT* | *CYGWIN_98* | *CYGWIN_95*)
+	    *win32* | *WIN32*)
 		    AC_MSG_RESULT(yes)
 		;;
 	    *)
@@ -444,7 +444,7 @@
 
 AC_DEFUN(SC_ENABLE_SYMBOLS, [
     case "`uname -s`" in
-	*win32* | *WIN32* | *CYGWIN_NT* | *CYGWIN_98* | *CYGWIN_95*)
+	*win32* | *WIN32*)
 	    tcl_dbgx=d
 	;;
 	*)
@@ -662,7 +662,19 @@
 	    LDFLAGS="-export-dynamic"
 	    LD_SEARCH_FLAGS=""
 	    ;;
-	*win32*|*WIN32*|CYGWIN_NT*|cygwin_nt*|*CYGWIN_98*|*CYGWIN_95*)
+	CYGWIN*)
+	    SHLIB_CFLAGS="-DDLL_EXPORT -DPIC"
+	    SHLIB_LD="${CC} -shared -Wl,--enable-auto-image-base"
+	    SHLIB_LD_LIBS='${LIBS}'
+	    SHLIB_SUFFIX=".dll"
+	    DL_OBJS="tclLoadDl.o"
+	    DL_LIBS=""
+	    LDFLAGS=""
+	    LD_SEARCH_FLAGS=""
+		TCL_NEEDS_EXP_FILE=1
+		TCL_EXPORT_FILE_SUFFIX='${VERSION}\$\{DBGX\}.exp'
+	    ;;
+	*win32*|*WIN32*)
 	    CFLAGS_DEBUG="-nologo -Z7 -Od -WX ${runtime}d"
 	    CFLAGS_OPTIMIZE="-nologo -Oti -Gs -GD ${runtime}"
 	    LDFLAGS_CONSOLE="-subsystem:console"
@@ -1805,7 +1817,7 @@
 
 AC_DEFUN(SC_MAKE_LIB, [
     case "`uname -s`" in
-	*win32* | *WIN32* | *CYGWIN_NT* |*CYGWIN_98*|*CYGWIN_95*)
+	*win32* | *WIN32*)
 	    if test "${CC-cc}" = "cl"; then
 		MAKE_STATIC_LIB="\${STLIB_LD} -out:\[$]@ \$(\[$]@_OBJECTS) "
 		MAKE_SHARED_LIB="\${SHLIB_LD} \${SHLIB_LDFLAGS} \${SHLIB_LD_LIBS} \$(LDFLAGS) -out:\[$]@ \$(\[$]@_OBJECTS) "
@@ -1892,7 +1904,7 @@
     done
 
     case "`uname -s`" in
-	*win32* | *WIN32* | *CYGWIN_NT* |*CYGWIN_98*|*CYGWIN_95*)
+	*win32* | *WIN32*)
 	    $1_LIB_SPEC=\"`${CYGPATH} ${$1_LIB_PATH_NAME}`\"
 	    ;;
 	*)
@@ -1938,7 +1950,7 @@
     AC_MSG_CHECKING(for Tcl private include files)
 
     case "`uname -s`" in
-	*win32* | *WIN32* | *CYGWIN_NT* |*CYGWIN_98*|*CYGWIN_95*)
+	*win32* | *WIN32*)
 	    TCL_TOP_DIR_NATIVE=\"`${CYGPATH} ${TCL_SRC_DIR}`\"
 	    TCL_GENERIC_DIR_NATIVE=\"`${CYGPATH} ${TCL_SRC_DIR}/generic`\"
 	    TCL_UNIX_DIR_NATIVE=\"`${CYGPATH} ${TCL_SRC_DIR}/unix`\"
@@ -2064,7 +2076,7 @@
     AC_MSG_CHECKING(for Tk private include files)
 
     case "`uname -s`" in
-	*win32* | *WIN32* | *CYGWIN_NT* |*CYGWIN_98*|*CYGWIN_95*)
+	*win32* | *WIN32*)
 	    TK_UNIX_DIR_NATIVE=\"`${CYGPATH} ${TK_SRC_DIR}/unix`\"
 	    TK_WIN_DIR_NATIVE=\"`${CYGPATH} ${TK_SRC_DIR}/win`\"
 	    TK_GENERIC_DIR_NATIVE=\"`${CYGPATH} ${TK_SRC_DIR}/generic`\"
@@ -2216,7 +2228,7 @@
     AC_CACHE_VAL(ac_cv_path_tclsh, [
 	search_path=`echo ${exec_prefix}/bin:${prefix}/bin:${TCL_BIN_DIR}:${TCL_BIN_DIR}/../bin:${PATH} | sed -e 's/:/ /g'`
 	for dir in $search_path ; do
-	    for j in `ls -r $dir/tclsh[[8-9]]*${EXEEXT} 2> /dev/null` \
+	    for j in `ls -r $dir/tclsh[[8-9]].*${EXEEXT} 2> /dev/null` \
 		    `ls -r $dir/tclsh*${EXEEXT} 2> /dev/null` ; do
 		if test x"$ac_cv_path_tclsh" = x ; then
 		    if test -f "$j" ; then
@@ -2260,7 +2272,7 @@
     AC_CACHE_VAL(ac_cv_path_wish, [
 	search_path=`echo ${exec_prefix}/bin:${prefix}/bin:${TCL_BIN_DIR}:${TCL_BIN_DIR}/../bin:${PATH} | sed -e 's/:/ /g'`
 	for dir in $search_path ; do
-	    for j in `ls -r $dir/wish[[8-9]]*${EXEEXT} 2> /dev/null` \
+	    for j in `ls -r $dir/wish[[8-9]].*${EXEEXT} 2> /dev/null` \
 		    `ls -r $dir/wish*${EXEEXT} 2> /dev/null` ; do
 		if test x"$ac_cv_path_wish" = x ; then
 		    if test -f "$j" ; then
